superset/db_engine_specs/elasticsearch.py

```diff
@@ -46,6 +46,7 @@ class ElasticSearchEngineSpec(BaseEngineSpec):  # pylint: disable=abstract-metho
         TimeGrain.MINUTE: "HISTOGRAM({col}, INTERVAL 1 MINUTE)",
         TimeGrain.HOUR: "HISTOGRAM({col}, INTERVAL 1 HOUR)",
         TimeGrain.DAY: "HISTOGRAM({col}, INTERVAL 1 DAY)",
+        TimeGrain.WEEK: "DATE_TRUNC('week', {col})",
```

Comment:
Why are we using `DATE_TRUNC` and not `HISTOGRAM`? Ideally it would be great if these were consistent.

Reply:
The `HISTOGRAM` groups the data by `INTERVAL` but [there is no "week" interval](https://www.elastic.co/guide/en/elasticsearch/reference/current/sql-functions-datetime.html#sql-functions-datetime-interval) in Elasticsearch. 

Perhaps there is a refactor needed to use `DATE_TRUNC` for the other time grains as well? I could do it, but as I am new to this community, could you let me know if I should do it in this PR or open a separate one for refactoring? 😄

Reply:
@mikelv92 thanks for the explanation. I think it's ok to merge this PR as is, but if you're interested in doing a fast follow to change all the existing time grains to use `DATE_TRUNC`—which _seems_ to be the [blessed option](https://www.elastic.co/guide/en/elasticsearch/reference/current/sql-functions-datetime.html#sql-functions-datetime-trunc)—that would be great.

Reply:
@john-bodley Here is the follow-up PR: #25717